src/tests/stats/test_distribution_uniform.fypp

```diff
@@ -0,0 +1,281 @@
+#:include "common.fypp"
+#:set ALL_KINDS_TYPES = INT_KINDS_TYPES + REAL_KINDS_TYPES + CMPLX_KINDS_TYPES
+program test_distribution_uniform
+    use stdlib_error, only : check
+    use stdlib_kinds, only: int8, int16, int32, int64, sp, dp, qp
+    use stdlib_stats_distribution_PRNG, only : random_seed, dist_rand
+    use stdlib_stats_distribution_uniform,                                      &
+        uni_rvs => uniform_distribution_rvs,                                    &
+        uni_pdf => uniform_distribution_pdf,                                    &
+        uni_cdf => uniform_distribution_cdf
+
+    implicit none
+    logical ::  warn = .true.
+    #:for k1, t1 in REAL_KINDS_TYPES
+    ${t1}$, parameter :: ${k1}$tol = 1000 * epsilon(1.0_${k1}$)
+    #:endfor
+    integer :: put, get
+
+    put = 135792468
+
+    call test_shuffle
+
+    call test_uni_rvs_0
+
+    #:for k1, t1 in ALL_KINDS_TYPES
+    call test_uni_rvs_${t1[0]}$${k1}$
+    #:endfor
+
+    #:for k1, t1 in ALL_KINDS_TYPES
+    call test_uni_pdf_${t1[0]}$${k1}$
+    #:endfor
+
+    #:for k1, t1 in ALL_KINDS_TYPES
+    call test_uni_cdf_${t1[0]}$${k1}$
+    #:endfor
+    stop
+
+    contains
+
+    subroutine test_shuffle
+        integer :: n(10), na(10) = [10, 6, 9, 2, 8, 1, 3, 5, 7, 4]
+        real :: x(10),  xa(10) = [5.0, 10.0, 9.0, 4.0, 3.0, 8.0, 2.0, 1.0,       &
+                                 7.0, 6.0]
+        complex :: z(10), za(10) = [(8.0, 8.0), (7.0, 7.0), (4.0, 4.0),           &
+                                  (1.0, 1.0), (5.0, 5.0), (9.0, 9.0),           &
+                                  (6.0, 6.0), (3.0, 3.0), (2.0, 2.0),           &
+                                  (10.0, 10.0)]
+        integer :: i, put, get
+
+        do i=1, 10
+            n(i) = i
+            x(i) = real(i)
+            z(i) = cmplx(real(i),real(i))
+        end do
+        put = 32165498
+        call random_seed(put, get)
+        n(:) = shuffle(n)
+        x(:) = shuffle(x)
+        z(:) = shuffle(z)
+        call check(all(n == na), &
+             msg="Integer shuffle failed test", warn=warn)
+        call check(all(x == xa), &
+             msg="Real shuffle failed test", warn=warn)
+        call check(all(z == za), &
+             msg="Complex shuffle failed test", warn=warn)
+    end subroutine test_shuffle
+
+    subroutine test_uni_rvs_0
+        integer :: i, j, freq(0:1000), num=10000000
+        real(dp) :: chisq, expct
+
+        print *,""
+        print *, "Test uniform random generator with chi-squared"
+        freq = 0
+        do i = 1, num
+            j = 1000 * uni_rvs( )
+            freq(j) = freq(j) + 1
+        end do
+        chisq = 0.0_dp
+        expct = num / 1000
+        do i = 0, 999
```

Comment:
```suggestion
        do i = 0, size_array -1
```
Why not `do i=0, size_array`?

Reply:
It is related to freedom of degree (f. of d.)for probability test. It is easy to look up table and find critical value for f. of d. = 1000, but not for 1001. The `freq` array holds 1000 bins (i.e., f. of d.) for random numbers generated. Since `freq` has to consider numbers around `0`, it is `0` based array.